pkg/utils/validator/clusterresourceplacement_test.go

```diff
@@ -174,23 +179,28 @@ func TestValidateClusterResourcePlacementAlpha(t *testing.T) {
 			},
 			resourceInformer: MockResourceInformer{},
 			wantErr:          true,
+			wantErrMsg:       "for 'in', 'notin' operators, values set can't be empty",
 		},
 	}
 	for testName, testCase := range tests {
 		t.Run(testName, func(t *testing.T) {
 			ResourceInformer = testCase.resourceInformer
-			if err := ValidateClusterResourcePlacementAlpha(testCase.crp); (err != nil) != testCase.wantErr {
-				t.Errorf("ValidateClusterResourcePlacementAlpha() error = %v, wantErr %v", err, testCase.wantErr)
+			gotErr := ValidateClusterResourcePlacementAlpha(testCase.crp)
+			if (gotErr != nil) != testCase.wantErr {
+				t.Errorf("ValidateClusterResourcePlacementAlpha() error = %v, wantErr %v", gotErr, testCase.wantErr)
+			}
+			if testCase.wantErr && !strings.Contains(gotErr.Error(), testCase.wantErrMsg) {
+				t.Errorf("ValidateClusterResourcePlacementAlpha() failed to find expected error message = %s, in error = %s", testCase.wantErrMsg, gotErr.Error())
```

Comment:
```suggestion
				t.Errorf("ValidateClusterResourcePlacementAlpha() got error %v, want contains %s",  gotErr, testCase.wantErrMsg)
```

you don't need to use %s,

"4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any)."

https://pkg.go.dev/fmt#hdr-Printing